src/Uno.UI/UI/Xaml/Media/Imaging/RenderTargetBitmap.cs

```diff
@@ -14,14 +14,89 @@
 using Uno.UI.Xaml.Media;
 using Buffer = Windows.Storage.Streams.Buffer;
 using System.Buffers;
+using System.Runtime.InteropServices;
 
 namespace Microsoft.UI.Xaml.Media.Imaging
 {
 #if NOT_IMPLEMENTED
 	[global::Uno.NotImplemented("IS_UNIT_TESTS", "__WASM__", "__NETSTD_REFERENCE__")]
 #endif
-	public partial class RenderTargetBitmap : ImageSource, IDisposable
+	public partial class RenderTargetBitmap : ImageSource
 	{
+#if !__ANDROID__
+		// This is to avoid LOH array allocations
+		private unsafe struct UnmanagedArrayOfBytes : IDisposable
+		{
+			public nint Pointer;
+			public int Length { get; }
+
+			public UnmanagedArrayOfBytes(int length)
+			{
+				Length = length;
+				Pointer = Marshal.AllocHGlobal(length);
```

Comment:
We'll likely need to [increase memory pressure](https://learn.microsoft.com/en-us/dotnet/api/system.gc.addmemorypressure?view=net-8.0) as a result of this allocation. (same foes for the dispose).